modules/gateway/gateway.go

```diff
@@ -6,17 +6,18 @@ import (
 	"net"
 	"os"
 	"path/filepath"
-	"time"
+	"sync"
 
 	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/persist"
-	"github.com/NebulousLabs/Sia/sync"
+	siasync "github.com/NebulousLabs/Sia/sync"
 )
 
 var (
 	errNoPeers     = errors.New("no peers")
 	errUnreachable = errors.New("peer did not respond to ping")
+	errClosed      = errors.New("Gateway is closed")
```

Comment:
probably better to use the error from the ThreadGroup, and then let the higher level process specify the module.